testplan/web_ui/web_app.py

```diff
@@ -94,20 +92,21 @@ def get(self, report_uid, assertions_uid):
 class TestplanAttachment(Resource):
     def get(self, report_uid, attachment_path):
         """Get an attachment for a specific Testplan report given their uids."""
-        attachment_path = os.path.abspath(
-            os.path.join(
-                app.config["DATA_PATH"], defaults.ATTACHMENTS, attachment_path
-            )
+        base_path = os.path.join(app.config["DATA_PATH"], defaults.ATTACHMENTS)
+        normlised_attachment_path = os.path.normpath(
+            os.path.join(base_path, attachment_path)
         )
-
-        if os.path.exists(attachment_path):
-            return send_from_directory(
-                directory=os.path.dirname(attachment_path),
-                filename=os.path.basename(attachment_path),
-            )
-        else:
+        if not (
+            normlised_attachment_path.startswith(base_path)  # security check
```

Comment:
Why are we doing normpath and security check here?

Reply:
because `attachment_path` can contain substring like `../` which will cause security issues, and it is detected by CodeQL scanning.